profiles/fscloud/variables.tf

```diff
@@ -0,0 +1,206 @@
+##############################################################################
+# Common variables
+##############################################################################
+
+variable "resource_group_id" {
+  type        = string
+  description = "The resource group ID where resources will be provisioned."
+}
+
+##############################################################################
+# COS instance variables
+##############################################################################
+
+variable "create_cos_instance" {
+  description = "Set as true to create a new Cloud Object Storage instance."
+  type        = bool
+  default     = true
+}
+
+variable "create_hmac_key" {
+  description = "Set as true to create a new HMAC key for the Cloud Object Storage instance."
+  type        = bool
+  default     = false
+}
+
+variable "hmac_key_name" {
+  description = "The name of the hmac key to be created."
+  type        = string
+  default     = "hmac-cos-key"
+}
+
+variable "hmac_key_role" {
+  description = "The role you want to be associated with your new hmac key. Valid roles are 'Writer', 'Reader', 'Manager', 'Content Reader', 'Object Reader', 'Object Writer'."
+  type        = string
+  default     = "Manager"
+}
+
+variable "cos_instance_name" {
+  description = "The name to give the cloud object storage instance that will be provisioned by this module. Only required if 'create_cos_instance' is true."
+  type        = string
+  default     = null
+}
+
+variable "cos_location" {
+  description = "Location to provision the cloud object storage instance. Only used if 'create_cos_instance' is true."
+  type        = string
+  default     = "global"
```

Comment:
Any point in having this as a variable? If we do, then we should add validation to only allow accepted values (which I think is only global?)